src/Sylius/Bundle/UserBundle/Form/EventListener/GuestCustomerFormListener.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UserBundle\Form\EventListener;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\User\Context\CustomerContextInterface;
+use Sylius\Component\User\Model\CustomerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class GuestCustomerFormListener implements EventSubscriberInterface
```

Comment:
Sorry for misleading you earlier, but we should call it `Subscriber` as the refactoring has already started (https://github.com/Sylius/Sylius/pull/2843) :smile: